zone/tradeskills.cpp

```diff
@@ -1866,37 +1866,59 @@ bool ZoneDatabase::DisableRecipe(uint32 recipe_id)
 bool Client::CheckTradeskillLoreConflict(int32 recipe_id)
 {
 	auto recipe_entries = TradeskillRecipeEntriesRepository::GetWhere(
-			content_db,
-			fmt::format(
-					"recipe_id = {} ORDER BY componentcount DESC",
-					recipe_id
-			)
+		content_db,
+		fmt::format(
+			"recipe_id = {} ORDER BY componentcount DESC",
+			recipe_id
+		)
 	);
 	if (recipe_entries.empty()) {
 		return false;
 	}
 
-	for (auto& f : recipe_entries) {
-		for (auto &e: recipe_entries) {
-			if (f.componentcount > 0 && e.item_id == f.item_id && e.componentcount == 0) {
-				e.item_id = 0;
+	const EQ::ItemData* f_item_inst = nullptr;
+	const EQ::ItemData* e_item_inst = nullptr;
+	for (auto &f : recipe_entries) {
+		if (f.item_id) {
+			f_item_inst = database.GetItem(f.item_id);
+			for (auto &e : recipe_entries) {
+				if (e.item_id && f_item_inst && f_item_inst->LoreGroup != 0) {
+					e_item_inst = database.GetItem(e.item_id);
+					if (
```

Comment:
This is insane to read, i'd like to see this broken up a bit more in bools that describe the intent of what's happening in this part to make it obvious to the next reader.

Example of local vars describing intent

```cpp
    for (int i = PlayerEvent::GM_COMMAND; i != PlayerEvent::MAX; i++) {
        bool is_in_database = std::find(db.begin(), db.end(), i) != db.end();
        bool is_deprecated  = Strings::Contains(PlayerEvent::EventName[i], "Deprecated");
        bool is_implemented = !Strings::Contains(PlayerEvent::EventName[i], "Unimplemented");
```

Imagine if that was all on one line or nested

Reply:
just retesting with these changes made, pretty sure I didn't mess anything up, but better to be safe.